src/web/app/src/components/DependenciesTable/Row.tsx

```diff
@@ -0,0 +1,233 @@
+import { useState } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import TableRow from '@mui/material/TableRow';
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
+import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
+import IconButton from '@mui/material/IconButton';
+import { Box, Collapse } from '@material-ui/core';
+import GitHubIcon from '@mui/icons-material/GitHub';
+import { VscIssues } from 'react-icons/vsc';
+import FolderIcon from '@mui/icons-material/Folder';
+import CopyrightIcon from '@mui/icons-material/Copyright';
+import TableCell from '@mui/material/TableCell';
+import { FiPackage } from 'react-icons/fi';
+
+import { dependencyDiscoveryUrl } from '../../config';
+
+const useStyles = makeStyles((theme) => ({
+  rowTitle: {
+    fontSize: '14px',
+    color: theme.palette.text.secondary,
+    fontWeight: 700,
+  },
+  rowTitleIcon: {
+    marginRight: '1rem',
+    fontSize: '2rem',
+  },
+  iconRow: {
+    textAlign: 'right !important' as 'right',
+  },
+  dependencyInfo: {
+    display: 'inline-flex',
+    alignItems: 'center',
+    margin: '1rem',
+  },
+  dependencyInfoIssue: {
+    margin: '1rem',
+    display: 'flex',
+    alignItems: 'center',
+  },
+  dependencyInfoIcon: {
+    fontSize: '2rem !important',
+    marginRight: '1.5rem',
+  },
+  issueCard: {
+    display: 'flex',
+    border: '1px',
+    borderColor: theme.palette.text.primary,
+    borderStyle: 'solid',
+    color: theme.palette.text.primary,
+    padding: '1rem',
+    marginTop: '1rem',
+    marginBottom: '1rem',
+  },
+  issueCardLink: {
+    textDecoration: 'none',
+  },
+  collapseBox: {
+    fontSize: '14px',
+    color: theme.palette.text.primary,
+    padding: '2rem',
+    fontFamily: 'Spartan',
+  },
+  issueCardTitle: {
+    fontWeight: 700,
+    fontFamily: 'Spartan',
+  },
+}));
+
+type issueInfo = {
+  htmlUrl: string;
+  title: string;
+  body: string;
+  createdAt: string;
+};
+
+type RowProps = {
+  dependency: string;
+};
+
+export const Row = ({ dependency }: RowProps) => {
+  const classes = useStyles();
+  const [open, setOpen] = useState<boolean>(false);
+  const [url, setUrl] = useState<string>('');
+  const [directory, setDirectory] = useState<string>('');
+  const [license, setLicense] = useState<string>('');
+  const [issues, setIssues] = useState<Array<issueInfo>>([]);
+  const [issueURL, setIssueUrl] = useState<string>('');
+  const [apiLimited, setApiLimited] = useState<boolean>(false);
+
+  // Fetching more info of the dependency
+  const fetchMoreDependencyInfo = async () => {
+    const projectInfo = await fetch(`${dependencyDiscoveryUrl}/projects/${dependency}`);
+    const { gitRepository: { url, directory } = { url: '', directory: '' }, license } =
+      await projectInfo.json();
+    setUrl(url);
+    setIssueUrl(
+      `${url}/issues?q=is%3Aopen+is%3Aissue+label%3A"good+first+issue"%2C"%27hacktoberfest"%2C"help+wanted"+`
```

Comment:
This is pretty dense.  Could you construct this encoded value from strings in an array instead, so it's easier to add/edit/read?

Reply:
Also, could this URL not get created in the backend and returned to you here as part of the JS we give you?

Reply:
I think it would be better to do it in our backend.

Reply:
Yes, I think it would be better in backend API to create this URL.

Reply:
Issue file, https://github.com/Seneca-CDOT/telescope/issues/3495

Reply:
https://github.com/Seneca-CDOT/telescope/pull/3496 waiting for this merge.